src/Crawler.php

```diff
@@ -74,6 +74,9 @@ class Crawler
     /** @var int */
     protected $delayBetweenRequests = 0;
 
+    /** @var string */
+    protected $userAgent = '*';
```

Comment:
Always use and modify the client's UA instead of managing a copy in the `Crawler` class.

Reply:
Fixed it. It now uses the client configuration as single source of truth and modifies the client user-agent with setUserAgent(). Added several tests also to verify this functionality.